sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/aio/client.py

```diff
@@ -72,11 +78,29 @@ def key_id(self) -> str:
         if not (self._key or self._get_key_forbidden):
             try:
                 self._key = await self._client.get_key(self._key_id.vault_url, self._key_id.name, self._key_id.version)
+                self._allowed_ops = frozenset(self._key.key_material.key_ops)
             except HttpResponseError as ex:
                 self._get_key_forbidden = ex.status_code == 403
         return self._key
```

Comment:
The name `_get_key_forbidden` seems pretty confusing here, since this function `get_key` will always return `self._key` regardless. I *think* `_get_key_forbidden` means that `self._client.get_key` should not be called?

Reply:
Correct, the intent is to prevent doomed `self._client.get_key` calls after we've tried to get the key and learned that's forbidden.

Reply:
Might suggest `_client_key_forbidden` as an alternative

Reply:
I'm trying on `_keys_get_forbidden`, which is confusing in its own way but more idiomatic because Key Vault calls the required permission 'keys/get'.